src/components/swapv2/LiquiditySourcesPanel/index.tsx

```diff
@@ -103,24 +91,81 @@ const SourceName = styled.span`
   color: ${({ theme }) => theme.text};
 `
 
+const LiquiditySourceHeader = styled.div`
+  border-top-right-radius: 8px;
+  border-top-left-radius: 8px;
+  background: ${({ theme }) => theme.tableHeader};
+  text-transform: uppercase;
+  font-size: 12px;
+  font-weight: 500;
+  padding: 12px;
+  color: ${({ theme }) => theme.subText};
+  display: flex;
+  gap: 1rem;
+  align-items: center;
+`
+
 const LiquiditySourcesPanel: React.FC<Props> = ({ onBack }) => {
-  const { chainId } = useActiveWeb3React()
   const [searchText, setSearchText] = useState('')
-  const debouncedSearchText = useDebounce(searchText.toLowerCase(), 200)
+  const debouncedSearchText = useDebounce(searchText.toLowerCase(), 200).trim()
 
-  const { data, error } = useAggregatorStats(chainId)
-  if (error || !data) {
-    onBack()
-    return null
-  }
+  const dexes = useAllDexes()
+  const [excludeDexes, setExcludeDexes] = useExcludeDexes()
 
-  const dexIDs = Object.keys(data.pools)
-  if (dexIDs.length === 0) {
-    onBack()
-    return null
-  }
+  const checkAllRef = useRef<any>()
+  const kyberSwapRef = useRef<any>()
 
-  const visibleDEXes = extractUniqueDEXes(dexIDs).filter(({ name }) => name.toLowerCase().includes(debouncedSearchText))
+  const kyberswapDexes = useMemo(() => {
+    return [
+      {
+        name: 'KyberSwap Elastic',
+        id: 'kyberswapv2',
+        logoURL: 'https://kyberswap.com/favicon.ico',
+      },
+      {
+        name: 'KyberSwap Classic',
+        id: 'kyberswapv1',
+        logoURL: 'https://kyberswap.com/favicon.ico',
+      },
+    ]
+  }, [])
+
+  useEffect(() => {
+    const selectedDexes = dexes?.filter(item => !excludeDexes.includes(item.id)) || []
+
+    if (selectedDexes.length === dexes?.length) {
+      checkAllRef.current.checked = true
```

Comment:
ref won't trigger re-renders, you should add some comments why they're used here

Reply:
Ah, it's ref of the input.

Reply:
yep, used for this kind of check box
<img width="39" alt="image" src="https://user-images.githubusercontent.com/11427213/185883019-09e6ab80-f18b-4eb9-bd17-687a456acb6e.png">